lang/cs/Org.Apache.REEF.IMRU/OnREEF/IMRUTasks/UpdateTaskHost.cs

```diff
@@ -148,27 +155,80 @@ public byte[] Call(byte[] memento)
                     "Received OperationCanceledException in UpdateTaskHost with message: {0}.",
                     e.Message);
             }
-            catch (IOException e)
+            catch (Exception e)
             {
-                Logger.Log(Level.Error, "Received IOException in UpdateTaskHost with message: {0}.", e.Message);
-                if (!_cancellationSource.IsCancellationRequested)
+                if (e is IOException || e is TcpClientConnectionException || e is RemotingException ||
```

Comment:
"The code tends to overcatch exceptions, usually catching the Exception class directly. Is such overcatching really necessary ?" We have to know the types of exceptions so that to decide which one is to recover and which one is not recoverable.